### PR TITLE
Refactor Profile Page

### LINKED ISSUE #191: Update profile page
## Description

Update the profile page based on Bryan's designs. Staff should be able to see why the user joined SSW, the user's submitted pitches, the user's claimed pitches, and potentially the user's unapproved claim requests. Note: some of these (specifically approved submitted pitches and claimed pitches) might be visible to any contributor who clicks on a user's profile page. Sahi will talk to Jason about this.

Also, add journalism experience and neighborhood user lives in to onboarding wizard and profile page. Staff should be able to see the journalism experience and neighborhood on any user's profile page. A user should be able to see this on their own profile page and edit both pieces.

Note: Whenever staff can do something, admin can also do what they do.

## Useful links

- Useful link

### PR DESCRIPTION
**Pull request by nkonjeti**
## Summary

Closes #191 
Add neighborhood dropdown and journalism experience fields in onboarding wizard. Change profile page to match Bryan's design. 

## Changes

- Added neighborhood dropdown and journalism experience input field 
- Added neighborhood and experience fields to User Schema (to test creating user with these new fields)
- Finished UI for profile page (including table and all of the fields)
- added user rating (not in model yet) -> has hardcoded rating w/ todo comment
- contributor/staff/admin can view other user's contributions
- staff/admin can view user reasoning for joining

## Screenshots
Created user with neighborhood and journalism experience 
image

image

image

Viewing your own profile
image
 
Admin viewing other user's profile
image

Contributor viewing other user's profile
image

FEEDBACK
image

### CODE REVIEW

File: client/src/components/FieldTag/index.tsx [hunk 1]
```diff
@@ -20,10 +20,6 @@ const FieldTag: FC<FieldTagProps> = ({
   ...rest
 }): ReactElement => {
   const toClassName = (str: string): string => {
-    if (name !== '') {
```
Comment (amit-sawhney) [inline client/src/components/FieldTag/index.tsx:23]: Why was this removed?

File: client/src/components/Lists/PitchItems/index.tsx [hunk 2]
```diff
@@ -0,0 +1,48 @@
+import React, { FC, ReactElement } from 'react';
+import { IPitch } from 'ssw-common';
+
+import { FieldTag } from '../..';
+import { useInterests, useTeams } from '../../../contexts';
+
+interface PitchItemsProps {
+  pitch: Partial<IPitch>;
```
Comment (amit-sawhney) [inline client/src/components/Lists/PitchItems/index.tsx:8]: Does this need to be a Partial?

File: client/src/components/Lists/PitchItems/index.tsx [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+  type: 'topics' | 'userTeams';
+}
+
+const PitchItems: FC<PitchItemsProps> = ({ pitch, type }): ReactElement => {
```
Comment (amit-sawhney) [inline client/src/components/Lists/PitchItems/index.tsx:12]: Andrew is writing a really similar component. Might want to cross check both and see which is better. The solution of just pulling the first assignment contributor's teams doesn't actually work. You could be on two teams (one of which is not needed on the pitch).

File: client/src/components/PitchStatus/index.tsx [hunk 4]
```diff
@@ -0,0 +1,23 @@
...
+
+interface PitchStatusProps {
+  date: Date;
+}
+
+const PitchStatus: FC<PitchStatusProps> = ({ date }): ReactElement => {
+  if (new Date(date) < new Date()) {
+    return (
+      <>
+        <FieldTag name="Published" hexcode="#E9F4E7" />
```
Comment (amit-sawhney) [inline client/src/components/PitchStatus/index.tsx:13]: Not the way to do this. Please use content and then add the stuff back to the FieldTag and then in the global styles, make a class for published-tag-color and in-progress-tag-color and assign these hexcodes.

File: client/src/components/PitchStatus/index.tsx [hunk 5]
```diff
@@ -0,0 +1,23 @@
...
+import { FieldTag } from '..';
+
+interface PitchStatusProps {
+  date: Date;
+}
+
+const PitchStatus: FC<PitchStatusProps> = ({ date }): ReactElement => {
+  if (new Date(date) < new Date()) {
+    return (
+      <>
```
Comment (amit-sawhney) [inline client/src/components/PitchStatus/index.tsx:12]: Why are these wrapped in JSX Fragment tags

File: client/src/components/PitchStatus/index.tsx [hunk 6]
```diff
@@ -0,0 +1,23 @@
+import React, { FC, ReactElement } from 'react';
+
+import { FieldTag } from '..';
+
+interface PitchStatusProps {
+  date: Date;
+}
+
+const PitchStatus: FC<PitchStatusProps> = ({ date }): ReactElement => {
```
Comment (amit-sawhney) [inline client/src/components/PitchStatus/index.tsx:9]: This is a Tag so please name it appropriately.

File: client/src/components/Tables/Contributions/index.tsx [hunk 7]
```diff
@@ -0,0 +1,41 @@
+import React, { FC, ReactElement } from 'react';
+import { IPitch } from 'ssw-common';
+
+import DynamicTable from '../DyanmicTable';
+
+import './styles.scss';
```
Comment (amit-sawhney) [inline client/src/components/Tables/Contributions/index.tsx:6]: Move to last import

File: client/src/components/Tables/Contributions/index.tsx [hunk 8]
```diff
@@ -0,0 +1,41 @@
...
+import {
+  dateColumn,
+  teamsColumn,
+  titleColumn,
+  topicsColumn,
+  statusColumn,
+} from './utils';
+
+interface ContributionsProps {
+  pitches: Partial<IPitch>[];
```
Comment (amit-sawhney) [inline client/src/components/Tables/Contributions/index.tsx:16]: Why Partial Pitches?

File: client/src/pages/profile/index.tsx [hunk 9]
```diff
@@ -80,375 +43,185 @@ const Profile = (): ReactElement => {
...
+
+        // pitches the user claimed a team for
+        const claimedPitches = res.data.result.aggregated.claimedPitches;
+
+        // filter assignment contributors to get teams claimed by user for pitch
+        claimedPitches.forEach((pitch) => {
+          pitch.assignmentContributors = pitch.assignmentContributors?.filter(
+            (contributor) => contributor.userId === userId,
+          );
+        });
```
Comment (amit-sawhney) [inline client/src/pages/profile/index.tsx:61]: First, why is this happening?
Second, why not just set every pitch.assignmentContributor array in the claimed picthes to be equal to `[userId]` if this is necessary. I am not sure why the claimed pitch assignment contributors need to change though.

File: client/src/pages/profile/index.tsx [hunk 10]
```diff
@@ -80,375 +43,185 @@ const Profile = (): ReactElement => {
...
-                onClick={startEditMode}
-                content="Edit Profile"
-              />
-            )}
-
-            <UserPicture size="small" user={user} />
+        <Grid.Row columns={5}>
+          <Grid.Column className="profile-pic-col" text-align="left" width={4}>
+            <div className="user-pic">
+              {' '}
```
Comment (amit-sawhney) [inline client/src/pages/profile/index.tsx:83]: Remove if not necessary

File: client/src/pages/profile/index.tsx [hunk 11]
```diff
@@ -80,375 +43,185 @@ const Profile = (): ReactElement => {
...
+              <div className="user-role">
+                <FieldTag content={user.role} />
+              </div>
+              <div className="rating">
+                {/* TODO: update when rating is added to model*/}
+                <Rating icon="star" defaultRating={3} maxRating={5} />
+                <p className="number-ratings">(16)</p>
+              </div>
+              <div>
+                {(userId === auth.user._id || auth.isAdmin) && (
```
Comment (amit-sawhney) [inline client/src/pages/profile/index.tsx:110]: This is the whole point of that isEditable thing and so only certain people could see certain information on the page. If you removed all of that, seems like some of the functionality of this page probably got deleted too.

File: client/src/pages/profile/index.tsx [hunk 12]
```diff
@@ -80,375 +43,185 @@ const Profile = (): ReactElement => {
...
-              <SelectAttribute
-                label={titleCase('role')}
-                value={titleCase(user.role)}
-                options={parseOptions(allRoles)}
-                onChange={(e, { value }) => editUser('role', `${value}`)}
-                viewable={isViewable('role')}
-                editable={isEditable('role')}
+          <Grid.Column width={3}>
+            <SocialsInput
+              disabled={user.email !== null}
```
Comment (amit-sawhney) [inline client/src/pages/profile/index.tsx:126]: consider `disabled={!user.email}`

File: client/src/pages/profile/index.tsx [hunk 13]
```diff
@@ -80,375 +43,185 @@ const Profile = (): ReactElement => {
...
+      <Divider />
+      {userId === auth.user._id ? (
+        <h2>Your Contributions</h2>
+      ) : (
+        <h2>
+          {`${
+            user.preferredName !== '' && user.preferredName
+              ? titleCase(user.preferredName)
+              : titleCase(user.firstName)
+          }'s` + ` Contributions`}
```
Comment (amit-sawhney) [inline client/src/pages/profile/index.tsx:209]: This needs to be abstracted.

**Review by amit-sawhney — CHANGES_REQUESTED** [review]
Overall good! The tags seem a little big across the board. Spacing could be improved between a lot of the tags. 

Check to make sure all of the functionality is still working. The big thing that isViewable and isEditable provided us with is that it didn't let nonAdmins or people other than yourself see the phoneNumber so keep that in mind.

File: client/src/components/FieldTag/index.tsx (hunk 1 above)
Comment (nkonjeti) [reply client/src/components/FieldTag/index.tsx:23]: For the field tags with pronouns I noticed when the name and content didn't match it didn't apply the styling for the tag to have the appropriate color. For example if the name was assigned "she/her" and the content was "pronouns", with this it would check if the name was not an empty string and assign the `className` to ''. I removed it so it wouldn't return '' in `toClassName` and apply the color to the tag using the content name.

File: client/src/components/Lists/PitchItems/index.tsx (hunk 2 above)
Comment (nkonjeti) [reply client/src/components/Lists/PitchItems/index.tsx:8]: The pitches (submitted and claimed) I got from IUserAggregate for the user were partial pitches.

File: client/src/components/Lists/PitchItems/index.tsx (hunk 3 above)
Comment (nkonjeti) [reply client/src/components/Lists/PitchItems/index.tsx:12]: I should probably make this component for only user teams because I use this when assignmentContributors is filtered to include only the user of the profile page, so it wouldn't work for other cases when assignmentContributors isn't filtered. In the profile page contributions table, for the column `teams you're on` we need to get the teams that the user had claimed for the pitch. I thought a way of doing that was from assignment contributors because it stores the teams for each user who is contributing. Wouldn't it have the teams that the user had picked for the pitch?

File: client/src/pages/profile/index.tsx (hunk 9 above)
Comment (nkonjeti) [reply client/src/pages/profile/index.tsx:61]: This was for the column in the contributions table `teams you're on`. I thought I could get this from assignment contributors because it stores the teams the user had been assigned to for the pitch. Since the table uses records of type pitches in the dynamic table I thought of passing this data through the pitches assignment contributors.

File: client/src/pages/profile/index.tsx (hunk 11 above)
Comment (nkonjeti) [reply client/src/pages/profile/index.tsx:110]: I had removed isEditable because I thought this would be abstracted into the edit profile modal. However, you're right I should keep isViewable for permissions on what the user can view.

**Review by bryange3 — CHANGES_REQUESTED** [review]
Looks good Neha! Minor comment: neighborhood, genders, pronouns, and races should all be the same style.

File: client/src/components/Lists/PitchItems/index.tsx (hunk 3 above)
Comment (nkonjeti) [reply client/src/components/Lists/PitchItems/index.tsx:12]: Never mind I'm going to this another way by filtering the user teams later on

**Review by bryange3 — APPROVED** [review]
Perfect Neha! Good work :)

**Review by amit-sawhney — APPROVED** [review]
